cachelib/allocator/CacheAllocator.h

```diff
@@ -2011,37 +2047,47 @@ class CacheAllocator : public CacheBase {
       }
     });
 
-    if (candidates.size() == 0) {
+    XDCHECK_EQ(candidates.size(),new_allocs.size());
+    if (candidates.size() < new_allocs.size()) {
+      throw std::runtime_error(
```

Comment:
Do you free the allocated memory before throwing the exception? Looks like a memory leak.